codegen/src/main/kotlin/org/jetbrains/bsp/generators/ir/IR.kt

```diff
@@ -59,21 +66,31 @@ sealed interface EnumType<A> {
 
 data class EnumValue<A>(val name: String, val value: A, val hints: List<Hint>)
 
-sealed interface Type {
-    data class Set(val member: Type) : Type
-    data class List(val member: Type) : Type
-    data class Map(val key: Type, val value: Type) : Type
-    data class Ref(val shapeId: ShapeId) : Type
+sealed interface InnerType {
+    data class Set(val member: Type) : InnerType
+    data class List(val member: Type) : InnerType
+    data class Map(val key: Type, val value: Type) : InnerType
+    object Ref : InnerType
```

Comment:
może zmieńmy Ref na Structure

Reply:
co prawda będzie to trochę dziwne, bo co to za typ "Struktura" (bez nazwy w dodatku). To znaczy nazwa oczywiście jest, ale w tej zewnętrznej klasie. Ale obawiam się, że nieważne jak zrobimy, to i tak będzie trochę dziwnie, bo mieszamy trochę rzeczy z różnych porządków

Reply:
głośno myślę jak coś, powiedz co Ty uważasz

Reply:
ja bym nie zmieniała tej nazwy, bo to czasami też jest Ref do enuma i to by trochę namieszało

Reply:
chociaż... na te enumy też byłoby spoko mieć osobne type żeby nie robić tego dziwnego rozdzielenia że open są typowani wbudowanymi, a closed jest Ref. Co o tym myślisz?

Reply:
można by na przykład zamienić `Ref` na `Struct` oraz dodać `IntEnum` i `StringEnum`

Reply:
brzmi w sumie jak spoko pomysł

Reply:
tylko do Type czy do IrShape?

Reply:
właśnie się zastanawiam, bo w sumie to chyba będzie bardziej pasować do IrShape jednak